src/azure-cli-core/azure/cli/core/local_context.py

```diff
@@ -56,47 +99,81 @@ def get(self, command, argument):
         return None
 
     def set(self, scopes, argument, value):
-        if self.is_on():
-            for scope in scopes:
-                self._local_context_file.set_value(scope, argument, value)
-
-    def is_on(self):
-        return self._local_context_file is not None
+        if self.is_on and self.current_dir:
+            if self.local_context_file is None:
+                file_path = os.path.join(self.current_dir, self.dir_name, self.file_name)
+                if not os.path.isfile(file_path):
+                    try:
+                        ensure_dir(os.path.dirname(file_path))
+                        with open(file_path, 'w'):
+                            pass
+                        os.chmod(file_path, stat.S_IRUSR | stat.S_IWUSR)
+                        self._load_local_context_file()
+                        logger.warning('Initiate local context in %s', self.current_dir)
+                    except Exception:  # pylint: disable=broad-except
+                        logger.warning('fail to set value to local context')
+            if self.local_context_file:
+                for scope in scopes:
+                    self.local_context_file.set_value(scope, argument, value)
 
     def turn_on(self):
-        current_dir = os.getcwd()
-        file_path = os.path.join(current_dir, self.dir_name, self.file_name)
-        if not os.path.isfile(file_path):
-            try:
-                ensure_dir(os.path.dirname(file_path))
-                with open(file_path, 'w'):
-                    pass
-                os.chmod(file_path, stat.S_IRUSR | stat.S_IWUSR)
-                self._local_context_file = None
-                self._load_local_context_file()
-            except Exception:  # pylint: disable=broad-except
-                raise CLIError('fail to turn on local context in {}.'.format(current_dir))
-            logger.warning('local context in %s is turned on.', current_dir)
-        else:
-            logger.warning('local context is already turned on in %s', current_dir)
+        self.config.set_value(LOCAL_CONTEXT_CONFIG_SECTION, self.username, 'on')
```

Comment:
Make sure it errors out if no user is logged in.

Reply:
Update the user to system account, it will have nothing to do with azure user.